### DISCUSSION TITLE
What CI Server is supported in the Reference Architecture? What if I use a different CI Server - how does that get handled?

### DISCUSSION THREAD

**Discussion by marijakstrazdas** [gruntwork-io/knowledge-base#242] [Help] [answered]
What CI Server is supported in the Reference Architecture? What if I use a different CI Server - how does that get handled?

**Comment by pete0emerson** [comment, marked as answer]
We currently support four CI servers:

* CircleCI
* GitLab
* Github Actions
* Jenkins

When a Reference Architecture is being provisioned for a customer, they decide which CI server they want to integrate with in their YAML form:

```
# TODO: Enter the CI server platform. Must be one of: circleci, jenkins, gitlab, github.
# This value indicates which type of CI configuration should be generated.
# NOTE: If you select Jenkins, you MUST select GitHub as the VCS platform. No other VCS is supported with Jenkins.
# Example:
#   CIServerPlatform: "circleci"
```

The integration with your CI server is a separate workflow from your application CI/CD pipeline. If you want to use a different CI server for your application than your infrastructure, you can do so.
